File: Problems/80/main.go
package main

func main() {}

func removeDuplicates(nums []int) int {
	k, two_cnt := 1, 1

	for i := 1; i < len(nums); i++ {
		if nums[i] != nums[i-1] {
			nums[k] = nums[i]
			k++
			two_cnt = 1
			continue
		}
		if two_cnt < 2 {
			nums[k] = nums[i]
			k++
			two_cnt++
		}
	}
	return k
}

File: Problems/80/main.cpp
#include <vector>
#include <iostream>
#include <string>
#include <map>
#include <stack>
#include <algorithm>
#include <numeric>
#include <unordered_set>
#include <unordered_map>
#include <queue>
#include <functional>
#include <sstream>
#include <set>

using namespace std;

class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        int n = nums.size();
        int k = 0, cnt = 0;

        for (int i = 1; i < n; i++)
            if (nums[i] != nums[k])
            {
                swap(nums[++k], nums[i]);
                cnt = 0;
            }
            else if (cnt == 0)
            {
                swap(nums[++k], nums[i]);
                cnt++;
            }

        return k + 1;
    }
};
